pkg/controller/controlplane/valuesprovider.go

```diff
@@ -516,12 +526,13 @@ func (vp *valuesProvider) getControlPlaneChartValues(
 	return values, nil
 }
 
-func (vp *valuesProvider) enableCSIADController() bool {
-	return vp.csi.EnableADController == nil || *vp.csi.EnableADController
+func (vp *valuesProvider) enableCSIADController(cpConfig *apisalicloud.ControlPlaneConfig) bool {
+	return vp.csi.EnableADController != nil && *vp.csi.EnableADController || cpConfig.CSI != nil && cpConfig.CSI.EnableADController != nil && *cpConfig.CSI.EnableADController
```

Comment:
So we should adjust the controller configuration and set `EnableADController` there to `false`. And let user decide whether they want this enabled for their old clusters.

Reply:
By default,  it is `false` if EnableADController  is not set in controller configuration.

Reply:
Yes, by default it is `false`. Just wondering if there is anywhere where we manually set it to `true`.